covariancematrix.go

```diff
@@ -25,8 +26,6 @@ func CovarianceMatrix(cov *mat64.Dense, x mat64.Matrix, wts []float64) *mat64.De
 
 	r, c := x.Dims()
 
-	// TODO(jonlawlor): indicate that the resulting matrix is symmetric, which
```

Comment:
I think we want to change it to *mat64.Symmetric later. In either event code is going to break, as it will be common to cast the returned matrix to an explicit type. In this case, we do want to promise the explicit type of matrix returned, because it will always be some form of dense matrix. The mat64.Matrix interface is great, but we need to be careful to not overuse it.